log-native/src/main/java/org/ballerinalang/stdlib/log/Utils.java

```diff
@@ -18,124 +18,106 @@
+                    if (value.equals("")) {
+                        value = "\"" + value + "\"";
+                    }
+                    break;
+                default:
+                    if (entry.getValue() instanceof BString) {
+                        value = "\"" + escape(entry.getValue().toString()) + "\"";
+                    } else {
+                        value = entry.getValue().toString();
+                    }
+                    break;
             }
+            message.append(key).append(" = ").append(value).append(" ");
         }
+        err.println(message);
     }
 
     /**
-     * Sets the global log level.
-     *
-     * @param logLevel log level
-     */
-    public static void setGlobalLogLevelExtern(BString logLevel) {
-        ballerinaUserLogLevel = BLogLevel.toBLogLevel(logLevel.getValue());
-        loggerLevels.put(GLOBAL_PACKAGE_PATH, BLogLevel.toBLogLevel(logLevel.getValue()));
-    }
-
-    /**
-     * Sets the module log level.
+     * Get the name of the current module.
      *
-     * @param module module
-     * @param logLevel log level
+     * @return module name
      */
-    public static void setModuleLogLevelExtern(BString module, BString logLevel) {
-        loggerLevels.put(module.getValue(), BLogLevel.toBLogLevel(logLevel.getValue()));
+    public static BString getModuleName() {
+        String className = Thread.currentThread().getStackTrace()[5].getClassName();
+        String[] pkgData = className.split("\\.");
+        if (pkgData.length > 1) {
+            String module = IdentifierUtils.decodeIdentifier(pkgData[1]);
+            return StringUtils.fromString(pkgData[0] + "/" + module);
+        }
+        return StringUtils.fromString(".");
     }
 
     /**
-     * Checks if module log level has been enabled.
+     * Get the current time.
      *
-     * @return true if module log level has been enabled, false if not.
+     * @return current time in yyyy-MM-dd HH:mm:ss.SSS format
      */
-    private static boolean isModuleLogLevelEnabled() {
-        return loggerLevels.size() > 1;
+    public static BString getCurrentTime() {
+        return StringUtils.fromString(
+                new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS")
+                        .format(new Date()));
     }
 
-    /**
-     * Get the log level of a given package.
-     *
-     * @param pkg package name
-     * @return the log level
-     */
-    private static BLogLevel getPackageLogLevel(String pkg) {
-        return loggerLevels.containsKey(pkg) ? loggerLevels.get(pkg) : ballerinaUserLogLevel;
+    private static String escape(String s) {
+        return s.replace("\\", "\\\\")
+                .replace("\t", "\\t")
+                .replace("\b", "\\b")
+                .replace("\n", "\\n")
+                .replace("\r", "\\r")
+                .replace("\f", "\\f")
+                .replace("'", "\\'")
+                .replace("\"", "\\\"");
```

Comment:
if log message already escaped the characters like `abcd\\t`, then what will be the log message.

Reply:
AFAIU, when a string value is assigned to a string variable, Ballerina unescapes that string value. Therefore the fields in the `LogRecord` record are always unescaped. So I think there won't be any escaped strings getting passed to `escape()` function at https://github.com/ballerina-platform/module-ballerina-log/blob/3408ba749ad9776d25d787403be1dd2847ed2052/log-native/src/main/java/org/ballerinalang/stdlib/log/Utils.java#L67